stl/inc/hash_map

```diff
@@ -220,11 +221,11 @@ namespace stdext {
             return *this;
         }
 
-        mapped_type& operator[](const key_type& _Keyval) {
+        _NODISCARD mapped_type& operator[](const key_type& _Keyval) {

```

Comment:
`operator[]` has side effects, so it shouldn't be marked as `_NODISCARD`. When `hm` doesn't already contain `key`, `hm[key]` inserts a key-value pair (with a value-initialized value), allowing `hm[key] = val;` to be used. (While `hm[key];` by itself is somewhat weird, it has well-defined behavior and real code might be doing that.) The Standard containers behave the same way:

https://github.com/microsoft/STL/blob/b1fbbdc4c31f0b810226e6eef782129e9e9c853c/stl/inc/map#L324
https://github.com/microsoft/STL/blob/b1fbbdc4c31f0b810226e6eef782129e9e9c853c/stl/inc/unordered_map#L383

(The design is indeed confusing, because `vector`'s `operator[]` is a pure observer.)

Reply:
> When hm doesn't already contain key, hm[key] inserts a key-value pair (with a value-initialized value), allowing hm[key] = val;

Ah, I didn't think about that 👍 .